### ISSUE TITLE
NoMethodError: undefined method 'supported?' for Proj4:Module

### ISSUE THREAD

**Issue by kurko** [rgeo/rgeo#192]
I'm getting a really weird issue and it looks like a bit of a rabbit hole.

### Background

When updating updating to Rails 5, I had to update `activerecord-postgis-adapter`, which now requires:

* `rgeo` 1.0.0 (previously 0.6.0)
* `proj4rb` 1.0.0 (previously 1.0.0, no changes)

### The issue

Running the tests now, I get this error:

``` 
  1) BusService.search respects text and coordinates same time
     Failure/Error:
       RGeo::Geographic.spherical_factory(
         srid: DEFAULT_SRID,
         proj4: 'omitted',
         coord_sys: wgs84_wkt
       )

     NoMethodError:
       undefined method `supported?' for Proj4:Module
     # /Users/kurko/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rgeo-1.0.0/lib/rgeo/coord_sys.rb:32:in `supported?'
     # /Users/kurko/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rgeo-1.0.0/lib/rgeo/coord_sys.rb:36:in `check!'
     # /Users/kurko/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rgeo-1.0.0/lib/rgeo/geographic/factory.rb:31:in `initialize'
     # /Users/kurko/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rgeo-1.0.0/lib/rgeo/geographic/interface.rb:127:in `new'
     # /Users/kurko/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rgeo-1.0.0/lib/rgeo/geographic/interface.rb:127:in `spherical_factory'
     # ./lib/rgeo_extensions.rb:104:in `default_factory'
     # ./spec/models/omitted_spec.rb:185:in `block (3 levels) in <top (required)>'
```

The respective [line](https://github.com/rgeo/rgeo/blob/996afc15d029ecf404bb2e537058897aeb59923a/lib/rgeo/coord_sys.rb#L39) causing the error is:

```rb
module CoordSys
  # The only valid key is :proj4
  def self.supported?(key)
    raise(Error::UnsupportedOperation, "Invalid key. The only valid key is :proj4.") unless key == :proj4
    defined?(Proj4) && Proj4.supported?
  end

  # ...
end
```

(996afc15d029ecf404bb2e537058897aeb59923a commit is where 1.0.0 is at)

I'm trying to get to the bottom of it, but no luck so far. Anyone have any hints?

Thanks

**Comment by teeparham** [comment]
`Proj4` was defined by another gem `proj4rb` (not `rgeo-proj4`, which provides `.supported?`). This is a bug & I will submit a fix soon. Thanks for reporting the issue.

**Comment by teeparham** [comment]
Fixed by 444d961d7fc. Released in version 1.1.1.

**Comment by kurko** [comment]
🎉  Thanks a lot
